### PR TITLE
Fix Search non-blocking follow-ups

### PR DESCRIPTION
**Pull request by KaiyuWang16**
## Summary of the Pull Request
This PR includes the non-blocking follow-ups of Search, which are addressed in this issue: https://github.com/microsoft/terminal/issues/3942

At current, the following two issues have been resolved: 

1. The place holder text in the TextBox ("Find...") needs to be localizable.
2. Search event should have a delegate with more args. This is illustrated below in Issue

The third issue, which is related to KeyDown events handling, is still under investigation. KeyDown events are now first sent to root TermControl, I need to figure out a way to make sure SearchBoxControl receives and handles the events first. 

 
## PR Checklist
* [x] Closes https://github.com/microsoft/terminal/issues/3942
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

### CODE REVIEW

File: src/cascadia/TerminalControl/SearchBoxControl.cpp [hunk 1]
```diff
@@ -30,26 +30,34 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
...
     {
-        return _goForward;
+        if (GoForwardButton().IsChecked().GetBoolean())
+        {
+            return true;
+        }
+        else
+        {
+            return false;
+        }
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalControl/SearchBoxControl.cpp]: can you not just `return GoForwardButton().IsChecked().GetBoolean()` just like `_CaseSensitive` below?

File: src/cascadia/TerminalControl/SearchBoxControl.cpp [hunk 2]
```diff
@@ -4,6 +4,7 @@
 #include "pch.h"
 #include "SearchBoxControl.h"
 #include "SearchBoxControl.g.cpp"
+#include <LibraryResources.h>
```
Comment (DHowett-MSFT) [inline src/cascadia/TerminalControl/SearchBoxControl.cpp]: since you are not using `RS_` you do not need this file

Comment (KaiyuWang16) [reply src/cascadia/TerminalControl/SearchBoxControl.cpp]: removed. I think I once used RS_ in this file but then moved it.

**Comment by KaiyuWang16** [comment]
> Don't we also have to localize Case Sensitive/Match Case, Find Up and Find Down? And Close?

Oh, you mean the tooltips ? I missed them, let me add them.

File: src/cascadia/TerminalControl/SearchBoxControl.cpp (hunk 1 above)
Comment (KaiyuWang16) [reply src/cascadia/TerminalControl/SearchBoxControl.cpp]: changed

File: src/cascadia/TerminalControl/SearchBoxControl.cpp [hunk 3]
```diff
@@ -67,14 +67,11 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
             auto const state = CoreWindow::GetForCurrentThread().GetKeyState(winrt::Windows::System::VirtualKey::Shift);
             if (WI_IsFlagSet(state, CoreVirtualKeyStates::Down))
             {
-                // We do not want the direction flag to change permanately
-                _goForward = !_goForward;
-                _SearchHandlers(*this, _textBox.Text());
-                _goForward = !_goForward;
+                _SearchHandlers(_textBox.Text(), !_GoForward(), _CaseSensitive());
```
Comment (zadjii-msft) [inline src/cascadia/TerminalControl/SearchBoxControl.cpp]: I am so much happier with this

File: src/cascadia/TerminalControl/SearchBoxControl.h [hunk 4]
```diff
@@ -28,29 +28,23 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
...
+        WINRT_CALLBACK(Search, SearchHandler);
         TYPED_EVENT(Closed, TerminalControl::SearchBoxControl, Windows::UI::Xaml::RoutedEventArgs);
 
     private:
-        bool _goForward; // The direction of the search, controlled by the buttons with arrows
-
-        winrt::Windows::UI::Xaml::Controls::Primitives::ToggleButton _goForwardButton;
-        winrt::Windows::UI::Xaml::Controls::Primitives::ToggleButton _goBackwardButton;
-        winrt::Windows::UI::Xaml::Controls::Primitives::ToggleButton _caseButton;
         winrt::Windows::UI::Xaml::Controls::TextBox _textBox;
```
Comment (zadjii-msft) [inline src/cascadia/TerminalControl/SearchBoxControl.h]: Hey if we're getting rid of the rest of these members, can we also replace `_textBox` with just `TextBox()`?

Comment (KaiyuWang16) [reply src/cascadia/TerminalControl/SearchBoxControl.h]: changed

**Comment by DHowett-MSFT** [comment]
I think you need to set `e.Handled(true)` in the searchbox ESC handler.

**Comment by KaiyuWang16** [comment]
> It looks like the ESC issue is not fixed. Was this commit supposed to fix the Esc issue?

Oh, I moved the handling of "Esc" to SearchBoxControl, but the early return check in TermControl::_KeyDownHandler is still necessary to avoid search dialog inputs being processed in TermControl.

**Comment by KaiyuWang16** [comment]
> I think you need to set e.Handled(true) in the searchbox ESC handler.

I am setting it as handled within the if(e.OriginalKey() == winrt::Windows::System::VirtualKey::Escape) condition. But setting all key inputs as handled will cause inputs into TextBox to be eaten......

**Comment by KaiyuWang16** [comment]
> I'm only blocking to make sure that point 3 in #3942 is addressed before this is merged. I'd also totally approve this PR as-is, so long as it doesn't mark that issue as closed. Everything in here so far is fine 😄

The point 3 is included in this new issue: https://github.com/microsoft/terminal/issues/4031

**Pull request merged**